### PR TITLE
Promql query APIs

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
Promql query APIs

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 1]
```diff
@@ -147,52 +149,124 @@ func parseSearchBody(jsonSource map[string]interface{}) (string, uint32, uint32,
...
+	segment.LogMetricsQuery("PromQL metrics query parser", &metricQueryRequest[0], qid)
+	res := segment.ExecuteMetricsQuery(&metricQueryRequest[0].MetricsQuery, &metricQueryRequest[0].TimeRange, qid)
 
-	searchText, startTime, endTime, step, _, err := parseSearchBody(readJSON)
+	mQResponse, err := res.GetResultsPromQl(&metricQueryRequest[0].MetricsQuery, pqlQuerytype)
 	if err != nil {
-		ctx.SetContentType(ContentJson)
-		ctx.SetStatusCode(fasthttp.StatusBadRequest)
-		_, err = ctx.WriteString(err.Error())
+		log.Errorf("ExecuteAsyncQuery: Error getting results! %+v", err)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:221]: nit: funcname is not correct, it sould be `ProcessPromqlMetricsSearchRequest: .. failed to executeAsyncQuery...`

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 2]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+func ProcessPromqlBuildInfoRequest(ctx *fasthttp.RequestCtx, myid uint64) {
+	buildInfo := map[string]interface{}{
+		"status": "success",
+		"data": map[string]string{
+			"version":   "2.23.1",
+			"revision":  "cb7cbad5f9a2823a622aaa668833ca04f50a0ea7",
+			"branch":    "master",
+			"buildUser": "julius@desktop",
+			"buildDate": time.Now().Format("20060102-15:04:05"),
+			"goVersion": runtime.Version(),
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:304]: hardcode the go version,

Comment (nkunal) [reply pkg/integrations/prometheus/promql/metricsSearchHandler.go:304]: define this `PromQLBuildInfoStr1 as a const string at the top of the file and use it directly at line number 315.
However write a quick UT, to parse this json const (just like what we are doing for promql functions json str

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 3]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+		log.Errorf("ProcessPromqlBuildInfoRequest: failed to write response, err=%v", err)
+	}
+}
+
+func ProcessGetLabelsRequest(ctx *fasthttp.RequestCtx, myid uint64) {
+	startParam := string(ctx.FormValue("start"))
+	endParam := string(ctx.FormValue("end"))
+	startTime, err := parseTimeForPromQL(startParam)
+	if err != nil {
+		log.Errorf("ProcessPromqlMetricsSearchRequest: Error parsing time parameter, err:%v", err)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:326]: function name not right

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 4]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	}
+	endTime, err := parseTimeForPromQL(endParam)
+	if err != nil {
+		log.Errorf("ProcessPromqlMetricsSearchRequest: Error parsing time parameter, err:%v", err)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:330]: function name not right

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 5]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	}
+
+	log.Printf("startTime: %v, local time: %v", startTime, time.Unix(int64(startTime), 0).Local())
+	log.Printf("endTime: %v, local time: %v", endTime, time.Unix(int64(endTime), 0).Local())
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:334]: remove the printfs

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 6]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	// TODO: Implement the logic to get the labels
+}
+
+func ProcessGetLabelValuesRequest(ctx *fasthttp.RequestCtx, myid uint64) {
+	labelName := utils.ExtractParamAsString(ctx.UserValue("labelName"))
+	startParam := string(ctx.FormValue("start"))
+	endParam := string(ctx.FormValue("end"))
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:341]: star and end are optional, do it accordingly

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 7]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	startTime, err := parseTimeForPromQL(startParam)
+	if err != nil {
+		log.Errorf("ProcessPromqlMetricsSearchRequest: Error parsing start time parameter, err:%v", err)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:344]: function name not right

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 8]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	}
+	endTime, err := parseTimeForPromQL(endParam)
+	if err != nil {
+		log.Errorf("ProcessPromqlMetricsSearchRequest: Error parsing end time parameter, err:%v", err)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:348]: function name not right

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 9]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	}
+
+	log.Printf("startTime: %v, local time: %v", startTime, time.Unix(int64(startTime), 0).Local())
+	log.Printf("endTime: %v, local time: %v", endTime, time.Unix(int64(endTime), 0).Local())
+	log.Printf("labelName: %v", labelName)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:353]: remobve printfs

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 10]
```diff
@@ -216,6 +290,93 @@ func ProcessMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+		log.Errorf("ProcessGetSeriesByLabelRequest: Error parsing 'start' parameter, err:%v", err)
+	}
+	endTime, err := parseTimeForPromQL(endParam)
+	if err != nil {
+		log.Errorf("ProcessGetSeriesByLabelRequest: Error parsing 'end' parameter, err:%v", err)
+	}
+
+	log.Printf("startTime: %v, local time: %v", startTime, time.Unix(int64(startTime), 0).Local())
+	log.Printf("endTime: %v, local time: %v", endTime, time.Unix(int64(endTime), 0).Local())
+	log.Printf("matches: %v", matches)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:379]: remove printds

**Review by Macbeth98 — APPROVED** [review]

**Pull request merged**